typing/oprint.ml

```diff
@@ -100,8 +100,21 @@ let print_out_value ppf tree =
     | Oval_nativeint i -> fprintf ppf "%nin" i
     | Oval_float f -> pp_print_string ppf (float_repres f)
     | Oval_char c -> fprintf ppf "%C" c
-    | Oval_string s ->
-        begin try fprintf ppf "%S" s with
+    | Oval_string (s, maxlen, kind) ->
+       begin try
+               let len = String.length s in
+                fprintf ppf "%s%S%s"
```

Comment:
You need parentheses for an expression in a currified context, but here, we are printing a data-structure with no function calls (except `Bytes.of_string`), no ?